src/core/lib/iomgr/ev_posix.h

```diff
@@ -85,16 +85,18 @@ typedef struct grpc_event_engine_vtable {
   void (*pollset_set_del_fd)(grpc_pollset_set* pollset_set, grpc_fd* fd);
 
   bool (*is_any_background_poller_thread)(void);
+  const char* name;
+  bool (*check_engine_available)(bool explicit_request);
```

Comment:
It seems odd to let the engine check whether or not the caller requested it. I'd suggest removing the argument and letting the caller decide whether or not it wants to check for availability.

Reply:
that'd be a semantic change to how we initialize this stuff... so future pr (maybe when we fork this stuff for ee)